buildSrc/src/main/kotlin/com/datadog/gradle/plugin/benchmark/ReviewBenchmarkResultsTask.kt

```diff
@@ -45,16 +46,31 @@ open class ReviewBenchmarkResultsTask : DefaultTask() {
 
     // region Internal
 
-    private fun processDeviceBenchmarks(deviceDir: File) {
+    private fun processBenchmarksPerDevice(
+        deviceDir: File
+    ) {
         println("processing benchmark results for device ${deviceDir.name}")
         if (deviceDir.exists() && deviceDir.isDirectory && deviceDir.canRead()) {
+            val benchmarksResults = mutableMapOf<String, Long>()
             deviceDir.listFiles()?.forEach {
                 if (it.isFile && it.canRead() && it.extension == "json") {
-                    check(visitor.visitBenchmarkJsonFile(it, extension.thresholds, extension.ignored)) {
-                        "One or more benchmark tests in ${it.path} didn't match the expected threshold."
-                    }
+                    visitor.visitBenchmarkJsonFile(it, benchmarksResults)
                 }
             }
+
+            val strategies = extension.benchmarkStrategies
+
+            // verify that all the benchmarks have at least one strategy
+            benchmarksResults.forEach {
+                checkNotNull(strategies[it.key]) {
```

Comment:
You can perform the check againt the strategy here, instead of doing another loop.